tests/unittests/tests-nanocoap/Makefile.include

```diff
@@ -1,2 +1,3 @@
 USEMODULE += nanocoap
 USEMODULE += nanocoap_token_ext
+USEMODULE += gnrc_sock # for sock_types.h a network stack is needed
```

Comment:
The test does not include that header nor does it use anything to do with `sock`, so why is it needed in this test?

Reply:
IMHO, if the fix needs a larger cleanup of `nanocoap`, the dependency should be moved to `nanocoap` for now, as this seems to be a `nanocoap` dependency issue, not a dependency issue of this unittest.

Reply:
That is because `#include "sock/udp.h"` is done in nanocoap. And that in turn has a `#include "sock_types.h"`, which is only found if any network stack is used, as this header is in the include dir of GNRC/lwIP/OpenWSN.

Reply:
> That is because `#include "sock/udp.h"` is done in nanocoap. And that in turn has a `#include "sock_types.h"`, which is only found if any network stack is used, as this header is in the include dir of GNRC/lwIP/OpenWSN.

I know, but as said in https://github.com/RIOT-OS/RIOT/pull/21190#discussion_r1942617078 this is a dependency issue of `nanocoap`, so it should be fixed there. I am just afraid that we forget to remove it once the dependency cleanup of the `nanocoap` parser is done, we forget to remove this incidental dependency.

Reply:
Is this approach better? (See new commits.)

Reply:
There is an issue that Murdock found, but that might just be a knock-on effect.